compiler_tester/src/test/case/input/deploy_evm.rs

```diff
@@ -110,6 +131,116 @@ impl DeployEVM {
         }
     }
 
+    ///
+    /// Runs the deploy transaction on native REVM.
+    ///
+    pub fn run_revm<'a, EXT, DB: Database>(
+        self,
+        summary: Arc<Mutex<Summary>>,
+        vm: revm::Evm<'a, EXT, State<DB>>,
+        mode: Mode,
+        test_group: Option<String>,
+        name_prefix: String,
+        evm_builds: &HashMap<String, Build, RandomState>,
+        evm_version: Option<EVMVersion>,
+    ) -> revm::Evm<'a, EXT, State<DB>> {
+        let name = format!("{}[#deployer:{}]", name_prefix, self.identifier);
+
+        let build = evm_builds
+            .get(self.identifier.as_str())
+            .expect("Always valid");
+        let mut deploy_code = build.deploy_build.bytecode.to_owned();
+        deploy_code.extend(self.calldata.inner.clone());
+
+        let vm = self.update_balance(vm);
+
+        let mut new_vm = vm
+            .modify()
+            .modify_env(|env| {
+                let evm_context = SystemContext::get_constants_evm(evm_version);
+                env.cfg.chain_id = evm_context.chain_id;
+                env.block.number = U256::from(evm_context.block_number);
+                let coinbase = web3::types::U256::from_str_radix(evm_context.coinbase, 16).unwrap();
+                env.block.coinbase = web3_u256_to_revm_address(coinbase);
+                env.block.timestamp = U256::from(evm_context.block_timestamp);
+                env.block.gas_limit = U256::from(evm_context.block_gas_limit);
+                env.block.basefee = U256::from(evm_context.base_fee);
+                let block_difficulty =
+                    web3::types::U256::from_str_radix(evm_context.block_difficulty, 16).unwrap();
+                env.block.difficulty = web3_u256_to_revm_u256(block_difficulty);
+                env.block.prevrandao = Some(B256::from(env.block.difficulty));
+                env.tx.gas_price = U256::from(0xb2d05e00_u32);
+                env.tx.gas_limit = evm_context.block_gas_limit;
+                env.tx.access_list = vec![];
+                env.tx.caller = web3_address_to_revm_address(&self.caller);
+                env.tx.data = revm::primitives::Bytes::from(deploy_code);
+                env.tx.value = revm::primitives::U256::from(self.value.unwrap_or_default());
+                env.tx.transact_to = TxKind::Create;
+            })
+            .build();
+
+        let res = match new_vm.transact_commit() {
+            Ok(res) => res,
+            Err(error) => {
+                match error {
+                    EVMError::Transaction(e) => {
+                        println!("Error on transaction: {:?}", e)
```

Comment:
We are not printing anything to std.
Please pass the error to `Summary::invalid` below instead.